### DISCUSSION TITLE
New Feature: Image Uploads

### DISCUSSION THREAD

**Discussion by steven-tey** [steven-tey/novel#37] [Announcements]
Novel just got a major upgrade: image uploads 📸

You can now drag and drop / paste images and they'll automatically be uploaded to the cloud.

https://github.com/steven-tey/novel/assets/28986134/61084252-579a-45c7-9322-2df76dab7b38

Image uploads & storage powered by @Vercel blob ▲

This works as a v1 of our image uploader, but the UX can be improved.

The ideal UX:
1. Show an instant base64 preview of the image
2. Add a loading indicator on the image as it uploads
3. Once upload is complete, swap out the URLs

I wrote about it here: https://github.com/steven-tey/novel/discussions/36

P.S.: Vercel Blob is still in private beta, so I made the `BLOB_READ_WRITE_TOKEN` optional – without it the images will be read & stored locally.

**Comment by DwayneSamuels** [comment]
🔥

**Comment by Favy-source** [comment]
I can't seem to successfully upload an image and save.
the BLOB_READ_WRITE_TOKEN not set msg keeps popping up
In my .env I have it as 'BLOB_READ_WRITE_TOKEN='
its empty. As its optional, can I just take it out? @steven-tey

**Comment by mathiasrscom** [comment]
I must be doing something wrong. As soon as I add the second image to the editor, my database stops responding in production. I looked at the json and figured out that the image had many characters after the bas64. Here is a shorten version: 

`{"alt":null,"src":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABTEAAAO/CAYAAAAXmYJrAAAABHNCSVQICAgIfAhkiAAAABl0RVh0U29mdHdhcmUAZ25vbWUtc2NyZWVuc2hvdO8Dvz4AAAArdEVYdENyZWF0aW9uIFRpbWUAdG9yIDEyIG9rdCAyMDIzIDE1O`

It might be too long for it to handle the request. Is there any way to change the upload API to use something else like Cloudinary and to just use URLs?
